polkadot/runtime/common/src/slots/mod.rs

```diff
@@ -221,6 +256,56 @@ pub mod pallet {
 			};
 			Ok(())
 		}
+
+		/// Try to refund the lease deposit before the actual end of the lease.
+		///
+		/// This is only allowed if parachain is in its last lease period and they started with more
+		/// than [`Config::MinLeasePeriodsForEarlyRefund`]. This is useful for parachains who want
+		/// to get access to their funds they used in the last lease and rebid using same for the
+		/// next lease.
+		///
+		/// Can only be called by [`Config::ForceOrigin`] or the Parachain manager.
+		#[pallet::call_index(3)]
+		#[pallet::weight(T::WeightInfo::early_lease_refund())]
+		pub fn early_lease_refund(origin: OriginFor<T>, para_id: ParaId) -> DispatchResult {
+			// ensure caller is ForceOrigin or the manager of the parachain.
+			ensure_signed(origin.clone())
+				.map_err(|e| e.into())
+				.and_then(|who| -> DispatchResult {
+					ensure!(
+						Some(who) == T::Registrar::manager_of(para_id),
+						Error::<T>::NoPermission
+					);
+					Ok(())
+				})
+				.or_else(|_| -> DispatchResult {
+					T::ForceOrigin::ensure_origin(origin)
+						.map(|_| ())
+						.map_err(|_| Error::<T>::NoPermission.into())
+				})?;
+
+			let min_lease_period_required = T::MinLeasePeriodsForEarlyRefund::get();
+			ensure!(
+				min_lease_period_required > 0u32.into(),
+				// if set to 0, we never allow early refund of slot deposit.
+				Error::<T>::PreconditionNotMet
```

Comment:
nit but maybe we want to distinguish this error and throw `Error::<T>::EarlyLeaseReturnNotPossible` instead.

Reply:
Created a new Error `EarlyRefundDisabled` to indicate that the feature is disabled.